core/src/ast_builder/expr/binary_op.rs

```diff
@@ -45,13 +45,25 @@ impl ExprNode {
         self.binary_op(BinaryOperator::Gt, other)
     }
 
+    pub fn gteq<T: Into<Self>>(self, other: T) -> Self {
+        self.binary_op(BinaryOperator::GtEq, other)
+    }
+
     pub fn lt<T: Into<Self>>(self, other: T) -> Self {
         self.binary_op(BinaryOperator::Lt, other)
     }
 
+    pub fn lteq<T: Into<Self>>(self, other: T) -> Self {
+        self.binary_op(BinaryOperator::LtEq, other)
+    }
+
     pub fn and<T: Into<Self>>(self, other: T) -> Self {
         self.binary_op(BinaryOperator::And, other)
     }
+
+    pub fn or<T: Into<Self>>(self, other: T) -> Self {
+        self.binary_op(BinaryOperator::Or, other)
+    }
```

Comment:
These would be enough to be as separate PR.
Supporting `and` is dependent to `filter` so `and` is supposed to be with `filter` improvement, but `gteq`, `lteq` and `or` are not related to `filter`.
Could you split and make another PR for supporting `gteq`, `lteq` and `or`?
It would be more common to use `gte` & `lte` than `gteq` & `lteq`.